lib/practice_exercises.rb

```diff
@@ -1,13 +1,65 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
+  unless string.class == String
+    raise ArgumentError, "input is not a string"
+  end 
+
+  a = string.split /[^A-Za-z]+/
+  b = a.map {|x| x.split //}
+  c = b.flatten 
+
+  i = 0 
+  j = c.length 
+  num = j/2
+
+  num.times do |n|
+    temp = []
+    temp << c[i]
+    temp << c[j-1]
+    if temp[0].downcase == temp[1].downcase
+      next
+    else 
+      return false
+    end 
+    i += 1
+    j -= 1
+  end 
+  
+  return true
+
   raise NotImplementedError, "Not implemented yet"
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) or O(n^2+n)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
This works, but I would say the time complexity is O(n * m) where m is the length of the shortest string.  Otherwise well done.